src/main/java/racing/domain/RacingGame.java

```diff
@@ -1,81 +1,40 @@
 package racing.domain;
 
 import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collections;
 import java.util.List;
+import java.util.stream.Collectors;
+import racing.domain.car.Car;
+import racing.domain.car.Cars;
+import racing.domain.car.Position;
+import racing.domain.startegy.MovingStrategy;
 
 public class RacingGame {
 
-    private static final int RANGE_OF_RANDOM_NUMBER = 10;
+    private final MovingStrategy movingStrategy;
+    private final Cars cars;
 
-    private final List<Car> cars;
-
-    public RacingGame() {
-        this.cars = new ArrayList<>();
-    }
-
-    public void startGame() {
-        try {
-            final List<String> nameList = UserInput.getCarNames();
-            for (String name : nameList) {
-                cars.add(new Car(name));
-            }
-            checkCarCount(cars);
-
-            final int tryNumber = UserInput.getTryNumber();
-            startRacing(tryNumber);
-            showGameResult();
-        } catch (IllegalArgumentException e) {
-            System.out.println(e.getMessage());
-            startGame();
-        }
-    }
-
-    private static void checkCarCount(List<Car> cars) {
-        if (cars.isEmpty()) {
-            throw new IllegalArgumentException("[ERROR] 최소 1대 이상의 자동차가 필요합니다.");
-        }
-    }
-
-    private void startRacing(final int tryNumber) {
-        System.out.println("\n실행 결과");
-        for (int i = 0; i < tryNumber; i++) {
-            driveCars();
-            System.out.println();
-        }
-    }
-
-    private void driveCars() {
-        for (Car car : cars) {
-            car.drive(generateRandomNumber());
-        }
+    public RacingGame(MovingStrategy movingStrategy, List<Car> cars) {
+        this.movingStrategy = movingStrategy;
+        this.cars = new Cars(cars);
     }
 
-    private int generateRandomNumber() {
-        return (int) (Math.random() * RANGE_OF_RANDOM_NUMBER);
+    public static RacingGame registerCars(MovingStrategy movingStrategy, List<String> names) {
+        return new RacingGame(movingStrategy,
+            names.stream().map(Car::new).collect(Collectors.toList()));
```

Comment:
가독성을 위해서 변수로 추출해보면 어떨까요? 🤗 
```suggestion
        List<Car> cars = names.stream()
                .map(Car::new)
                .collect(Collectors.toList());
        return new RacingGame(movingStrategy, cars);
```

Reply:
앗! 넵! 변수로 추출하는 방식으로 변경하겠습니다!